requests_mock/response.py

```diff
@@ -172,20 +172,20 @@ def create_response(request, **kwargs):
     headers = kwargs.pop('headers', {})
     encoding = None
 
-    if content is not None and not isinstance(content, six.binary_type):
-        raise TypeError('Content should be binary data')
-    if text is not None and not isinstance(text, six.string_types):
-        raise TypeError('Text should be string data')
-
-    if json is not None:
-        encoder = kwargs.pop('json_encoder', None) or jsonutils.JSONEncoder
-        text = jsonutils.dumps(json, cls=encoder)
-    if text is not None:
-        encoding = get_encoding_from_headers(headers) or 'utf-8'
-        content = text.encode(encoding)
-    if content is not None:
-        body = _IOReader(content)
     if not raw:
+        if content is not None and not isinstance(content, six.binary_type):
+            raise TypeError('Content should be binary data')
+        if text is not None and not isinstance(text, six.string_types):
+            raise TypeError('Text should be string data')
+
+        if json is not None:
+            encoder = kwargs.pop('json_encoder', None) or jsonutils.JSONEncoder
+            text = jsonutils.dumps(json, cls=encoder)
+        if text is not None:
+            encoding = get_encoding_from_headers(headers) or 'utf-8'
+            content = text.encode(encoding)
+        if content is not None:
+            body = _IOReader(content)
```

Comment:
It's not right or wrong - but I don't want to change this block for this purpose.

The whole logic here kind of falls through to the next step based on what's been set and whilst we could reorganize it a little for efficiency it's true for all these statements, not just raw. 

Unless i'm missing something please revert this bit.